intro/README.md

```diff
@@ -0,0 +1,561 @@
+2. [Dynamic Arrays](https://dlang.org/spec/arrays.html#dynamic-arrays): their length may change dynamically at run time.
+
+### Slicing
+
+Slicing an array means to specify a subarray of it.
+An array slice does not copy the data, it is only another reference to it:
+
+```d
+void main()
+{
+    int[10] a;       // declare array of 10 ints
+    int[] b;
+ 
+    b = a[1..3];     // a[1..3] is a 2 element array consisting of a[1] and a[2]
+    int x = b[1];    // equivalent to `int x = 0;`
+    a[2] = 3;
+    int y = b[1];    // equivalent to `int y = 3;`
+}
+```
+
+### Array operations
+
+1. Array setting:
+
+    ```d
+    void main()
+    {
+        int[3] a = [1, 2, 3];
+        int[3] c = [ 1, 2, 3, 4];  // error: mismatched sizes
+        int[] b = [1, 2, 3, 4, 5];
+        a = 3;                     // set all elements of a to 3 
+        b = 5;                     // set all elements of b to 5
+        a[] = 2;                   // same as `a = 3`; using an empty slice is the same as slicing the full array
+        b = a[0 .. $];             // $ evaluates to the length of the array (in this case 10)
+        b = a[];                   // semantically equivalent to the one above
+        b = a[0 .. a.length]       // semantically equivalent to the one above
+        b[] = a[];                 // semantically equivalent to the one above
+        b[2 .. 4] = 4              // same as b[2] = 4, b[3] = 4
+        b[0 .. 4] = a[0 .. 4]      // error, a does not have 4 elements
+        a[0 .. 3] = b              // error, operands have different length
+    }
+    ```
+        
+    **.length** is a builtin array property.
+    For an extensive list of array properties click [here](https://dlang.org/spec/arrays.html#array-properties).
+
+    Now let's see how arrays work with functions.
+    The next quiz can be tricky but it will help us better understand how arrays work.
+    If it help, think what would happed in a similar situation in C.
+    
+    [Quiz](./quiz/ref-vs-value.md)
+    
+2. Array Concatenation:
+
+    ```d
+    void main()
+    {
+        int[] a;
+        int[] b = [1, 2, 3];
+        int[] c = [4, 5];
+     
+        a = b ~ c;       // a will be [1, 2, 3, 4, 5]
+        a = b;           // a refers to b
+        a = b ~ c[0..0]; // a refers to a copy of b
+        a ~= c;          // equivalent to a = a ~ c;
+    }
+    ```
+       
+    Concatenation always creates a copy of its operands, even if one of the operands is a 0-length array.
+    The operator `~=` does not always create a copy.
+    For more information click [here](https://dlang.org/spec/arrays.html#resize).
+
+    Navigate to **demo/slice** directory.
+    Inspect and run the file **slice.d**.
+    What happened here? Why?
+
+    [Quiz](./quiz/slice.md)
```

Comment:
You could add here the exercise to improve the initial example by using slices instead of the for loop.

Reply:
If you are talking about the exercise with binary search where they have to fix a bug, I feel like the part with the unittest should be first, because chances are they will fix the bug without even realizing, and I think the main purpose of that exercise is to show how useful unittests can be

Reply:
No, I was referring to the initial example where an element is deleted from an array.